ast/src/expression/binary.rs

```diff
@@ -16,33 +16,59 @@
 
 use super::*;
 
+/// A binary operator.
+///
+/// Precedence is defined in the parser.
 #[derive(Debug, Copy, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub enum BinaryOperation {
+    /// Addition, i.e. `+`.
     Add,
+    /// Subtraction, i.e. `-`.
     Sub,
+    /// Multiplication, i.e. `*`.
     Mul,
+    /// Division, i.e. `/`.
     Div,
+    /// Exponentiation, i.e. `**` in `a ** b`.
     Pow,
+    /// Logical-or, i.e., `||`.
```

Comment:
Could consider saying 'Logical disjunction' and 'Logical conjunction' to use nouns as in 'addition', 'subtraction', etc.